### PR TITLE
Aslr

### PR DESCRIPTION
**Pull request by jamill**
Port ASLR compile flags upstream

### CODE REVIEW

File: config.mak.uname [hunk 1]
```diff
@@ -556,11 +556,19 @@ else
 	ifeq ($(shell expr "$(uname_R)" : '2\.'),2)
 		# MSys2
 		prefix = /usr/
+		# Enable DEP
+		BASIC_LDFLAGS += -Wl,--nxcompat
+		# Enable ASLR (unless debugging)
+		ifneq (,$(findstring -O,$(CFLAGS)))
```
Comment (jamill) [inline config.mak.uname]: Should this be enabled only when optimization is not `-O0`?

Comment (dscho) [reply config.mak.uname]: The idea is to enable this only for optimized builds `-O0`, `-O2` or any other `-O<n>` option

File: config.mak.uname [hunk 2]
```diff
@@ -556,11 +556,19 @@ else
...
+			BASIC_LDFLAGS += -Wl,--dynamicbase
+		endif
 		ifeq (MINGW32,$(MSYSTEM))
 			prefix = /mingw32
+			BASIC_LDFLAGS += -Wl,--pic-executable,-e,_mainCRTStartup
```
[the remark this answers is not in the document]
Comment (dscho) [reply config.mak.uname]: They don't hurt. So I'd rather have them, to try to make the "for debugging" build as similar to the "for production" build as possible.

**Pull request closed without merging**